tools/src/test/java/org/apache/kafka/tools/consumer/TransactionLogMessageFormatterTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools.consumer;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.protocol.MessageUtil;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogKey;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogValue;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TransactionLogMessageFormatterTest {
+
+    private static TransactionLogKey txnLogKey;
+    private static TransactionLogValue txnLogValue;
+    private static final String TOPIC = "TOPIC";
+
+    @BeforeAll
+    public static void setUp() {
+        txnLogKey = new TransactionLogKey()
+                .setTransactionalId("TXNID");
+        txnLogValue = new TransactionLogValue()
+                .setProducerId(100)
+                .setProducerEpoch((short) 50)
+                .setTransactionStatus((byte) 4)
+                .setTransactionStartTimestampMs(750L)
+                .setTransactionLastUpdateTimestampMs(1000L)
+                .setTransactionTimeoutMs(500)
+                .setTransactionPartitions(emptyList());
+    }
+
+    private static Stream<Arguments> parameters() {
+        return Stream.of(
+                Arguments.of((short) 10, (short) 10,
```

Comment:
Could you add test for `null` key and `null` value?

Reply:
There are two tests`testNullPayLoadWithTransactionLogMessage` and `testNullKeyWithTransactionLogMessage` to test null key and null value